### PR TITLE
Wip/6.0 SelectEagerCollectionInitializer

### PR DESCRIPTION
**Pull request by dreab8**
Given an `Order` with a eager collection of `Item`s
the `session.createQuery( "from Order", Order.class ).uniqueResult();`
should not create one Sql query with a join but 2 different queries.

### CODE REVIEW

File: hibernate-core/src/test/java/org/hibernate/orm/test/onetomany/OneToManyEmptyCollectionTest.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+		final StatisticsImplementor statistics = scope.getSessionFactory().getStatistics();
+		statistics.clear();
+		scope.inTransaction(
+				session -> {
+					Order order = session.createQuery( "from Order", Order.class ).uniqueResult();
+					assertThat( statistics.getPrepareStatementCount(), is( 2L ) );
+					List<Item> items = order.getEagerItems();
+					assertThat( items.size(), is( 0 ) );
+					assertThat( items, instanceOf( PersistentBag.class ) );
+					assertThat( items, instanceOf( PersistentBag.class ) );
```
Comment (NathanQingyangXu) [inline hibernate-core/src/test/java/org/hibernate/orm/test/onetomany/OneToManyEmptyCollectionTest.java]: seems duplicated testing statement above

**Comment by sebersole** [comment]
What do you think about removing that `selected` variable and handling that concept in a different way?

**Comment by dreab8** [comment]
@sebersole I'm open to suggestion, it would be great to completely remove the `selected` variable

**Comment by dreab8** [comment]
merged

**Pull request closed without merging**